backend/open_webui/retrieval/utils.py

```
@@ -406,6 +532,248 @@ def process_query(collection_name, query):
     return merge_and_sort_query_results(results, k=k)
 
 
+def generate_openai_batch_embeddings(
+    model: str,
+    texts: list[str],
+    url: str = "https://api.openai.com/v1",
+    key: str = "",
+    prefix: str = None,
```

Ruff (0.14.7)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)